core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1223,7 +1225,7 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     reportExitValue(e)
 
     // ensure we notify observers in the same order they subscribed to us
-    val iterator = observers.reverse.iterator
+    val iterator = observers.reverseIterator
```

Comment:
👍 

I wonder even if this guarantee is required. Maybe it's sufficient to notify in reverse order.